hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/HoodieMergeHandle.java

```diff
@@ -273,7 +273,8 @@ public WriteStatus close() {
           insertRecordsWritten++;
         }
       }
-      keyToNewRecords.clear();
+
+      ((ExternalSpillableMap) keyToNewRecords).close();
```

Comment:
merge handle seems to be the place where  deleting on jvm shutdown would be a problem. In other cases, they are fairly long lived.